cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -490,15 +489,9 @@ public BrokerStats cachedBrokerLoadStats(boolean allowCapacityEstimation) {
     synchronized (this) {
       if (_cachedBrokerLoadGeneration != null
           && clusterGeneration == _cachedBrokerLoadGeneration.clusterGeneration()
-          && _partitionMetricSampleAggregator.generation() == _cachedBrokerLoadGeneration.loadGeneration()) {
-        if (!allowCapacityEstimation) {
-          // Ensure that there is no capacity estimation in the cached model.
-          for (SingleBrokerStats singleBrokerStats : _cachedBrokerLoadStats.stats()) {
-            if (singleBrokerStats.isEstimated()) {
-              return null;
-            }
-          }
-        }
+          && _partitionMetricSampleAggregator.generation() == _cachedBrokerLoadGeneration.loadGeneration()
+          && _cachedBrokerLoadStats != null
+          && (allowCapacityEstimation || !_cachedBrokerLoadStats.isBrokerStatsEstimated())) {
```

Comment:
Can we perform the checks that do not depend on the metadata update before the metadata update? This helps us avoid a potentially redundant metadata query.